modules/sor/sorV2/lib/pathGraph/pathGraph.ts

```diff
@@ -457,35 +456,29 @@ export class PathGraph {
         return filtered;
     }
 
-    private getLimitAmountSwapForPath(path: PathGraphEdgeData[], swapKind: SwapKind): bigint {
-        let limit = path[path.length - 1].pool.getLimitAmountSwap(
+    private getLimitAmountSwapForPath(path: PathGraphEdgeData[]): bigint {
+        let limit;
+
+        limit = path[path.length - 1].pool.getLimitAmountSwap(
             path[path.length - 1].tokenIn,
             path[path.length - 1].tokenOut,
-            swapKind,
+            SwapKind.GivenIn,
         );
 
         for (let i = path.length - 2; i >= 0; i--) {
-            const poolLimitExactIn = path[i].pool.getLimitAmountSwap(
-                path[i].tokenIn,
-                path[i].tokenOut,
-                SwapKind.GivenIn,
-            );
-            const poolLimitExactOut = path[i].pool.getLimitAmountSwap(
-                path[i].tokenIn,
-                path[i].tokenOut,
-                SwapKind.GivenOut,
-            );
-
-            if (poolLimitExactOut <= limit) {
-                limit = poolLimitExactIn;
+            let limitGivenIn;
+            let limitGivenOut;
+            limitGivenIn = path[i].pool.getLimitAmountSwap(path[i].tokenIn, path[i].tokenOut, SwapKind.GivenIn);
+            limitGivenOut = path[i].pool.getLimitAmountSwap(path[i].tokenIn, path[i].tokenOut, SwapKind.GivenOut);
+            if (limitGivenOut <= limit) {
+                limit = limitGivenOut;
```

Comment:
```suggestion
                limit = limitGivenIn;
```

Reply:
This should be givenOut, not givenIn, that's why the comparison in the if is with limitGivenOut, the old version was wrong

Reply:
Are you sure?
If I understood it correctly, this function is propagating the limit all the way to the tokenIn, so paths can be comparable. Which means that it should indeed be `limitGivenIn`.